### PR TITLE
workaround to make it work with low-level http-kit/request function;

### PR DESCRIPTION
**Pull request by timgluz**
Hi,

i had issues to run mocks when i used low-level request method directly; 

I added simple workaround and 2additional test-cases;

Timo

### CODE REVIEW

File: src/org/httpkit/fake.clj [hunk 1]
```diff
@@ -115,13 +115,13 @@
...
-           (map #(% orig-fn opts callback))
-           (keep identity)
-           (first)))))
+  ([spec]
+    (let [handlers (build-handlers spec)]
+      (fn [orig-fn opts & callback]
+        (->> handlers
+             (map #(% orig-fn opts (first callback)))
+             (keep identity)
+             (first))))))
```
Comment (d11wtq) [inline src/org/httpkit/fake.clj]: What is the purpose of this change?

File: README.md [hunk 2]
```diff
@@ -46,6 +46,18 @@ Use the `with-fake-http` macro to fake some HTTP responses.
                  #"^http://localhost/" :allow]
   (http/get "http://localhost/foo")     ; <<some real response>> (:allow)
   (http/get "http://localhost/unsafe")) ; IllegalArgumentException (:deny)
+  
+;; using low-level http-kit/request method with older http-fake(<0.2.2)
+;; requires explicit callback function for workaround
+(with-fake-http ["http://foo.bar" "ok"]
```
Comment (d11wtq) [inline README.md:52]: Your indentation-style is not following mine. Would you mind updating it so the body expressions are only indented by 2 spaces?

File: test/org/httpkit/fake_test.clj [hunk 3]
```diff
@@ -177,4 +177,20 @@
...
+                                       (fn [req] req)))))))
+
+     (testing "http-kit/request works correctly without callback function. "
+      (with-fake-http ["http://foo.com/" "ok"]
+          (is (= "ok"
+                 (:body @(http/request {:url "http://foo.com/" :method :get}))))))
+
+ 
+
+  ;; end of test
```
Comment (d11wtq) [inline test/org/httpkit/fake_test.clj]: Could you please remove all the whitespace and the "end of test" comment? :) The file should end with a string of closing parentheses.

File: test/org/httpkit/fake_test.clj [hunk 4]
```diff
@@ -177,4 +177,20 @@
...
+                 (:body @(http/get "http://bar.co/")))))))
+    
+    (testing "http-kit/request works correctly when callback-function is defined"
+      (with-fake-http ["http://foo.com/" "ok"]
+          (is (= "ok"
+                 (:body @(http/request {:url "http://foo.com/" :method :get}
+                                       (fn [req] req)))))))
+
+     (testing "http-kit/request works correctly without callback function. "
+      (with-fake-http ["http://foo.com/" "ok"]
```
Comment (d11wtq) [inline test/org/httpkit/fake_test.clj:189]: Somehow this is only indented by one space, instead of two. What editor are you using?

File: test/org/httpkit/fake_test.clj [hunk 5]
```diff
@@ -177,4 +177,20 @@
 
         (testing "allows matched urls"
           (is (= "ok"
-                 (:body @(http/get "http://bar.co/")))))))))
+                 (:body @(http/get "http://bar.co/")))))))
+    
+    (testing "http-kit/request works correctly when callback-function is defined"
+      (with-fake-http ["http://foo.com/" "ok"]
+          (is (= "ok"
+                 (:body @(http/request {:url "http://foo.com/" :method :get}
```
Comment (d11wtq) [inline test/org/httpkit/fake_test.clj]: This is indented too deeply. Inside a macro body should only be indented by 2 spaces. You should be able to configure your editor to do this for anything starting with `with`, although that is typically a default in editors that support lisps.

**Comment by d11wtq** [comment]
Thank you for bringing this to my attention and for sending the PR. I've left a few comments on the diff that will need reviewing before I can merge it. You should generally never send a PR from your 'master' branch too.

File: test/org/httpkit/fake_test.clj (hunk 4 above)
Comment (timgluz) [inline test/org/httpkit/fake_test.clj:189]: I'm using Vim and Vim-Clojure - sometimes, specially when parenthesis are not closed yet, then autoident uses 1space;

**Comment by timgluz** [comment]
i did some style fixes. Is there anything else required? 
When can you release new version? I'm currently using my own localrepo, but i'm ready to publish my library and i need this fix  to pass tests on travis.

**Comment by d11wtq** [comment]
Great, thanks! I'll push a release to clojars now.

**Comment by timgluz** [comment]
Thanks!

**Comment by d11wtq** [comment]
0.2.2 released. Thank you!
